### PR TITLE
Fully support a custom access log format

### PR DESCRIPTION
**Pull request by anfleene**

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Overrides of opsworks nginx attributes.
+# gi
```
Comment (bbergstrom) [inline attributes/default.rb]: gi?

**Comment by bbergstrom** [comment]
Maybe an addition to the tests to confirm the log format gets written as expected?

File: attributes/default.rb (hunk 1 above)
Comment (anfleene) [inline attributes/default.rb]: hah vim fail

File: test/integration/maintenance/serverspec/nginx_spec.rb [hunk 2]
```diff
@@ -9,11 +9,6 @@
   its(:content) { should match /set \$maintenance 1;/}
 end
 
-describe command('curl -i localhost/system/maintenance.html') do
```
Comment (anfleene) [inline test/integration/maintenance/serverspec/nginx_spec.rb]: I gave up on this test I have no idea why it wasn't working. I logged into the box and it did exactly what the test is expecting it to do :disappointed:

**Comment by anfleene** [comment]
Good call.

File: templates/default/nginx.conf.erb [hunk 3]
```diff
@@ -23,7 +23,7 @@ http {
 <% end %>
 <% end %>
 
-  access_log	<%= node[:nginx][:log_dir] %>/access.log;
+access_log	<%= node[:nginx][:log_dir] %>/access.log<%=" #{node[:nginx][:log_format_default]}" unless node[:nginx][:log_format_default].nil? %>;
```
Comment (bbergstrom) [inline templates/default/nginx.conf.erb]: Your attribute addition is `log_format` not `log_format_default`. Should that match?

**Comment by anfleene** [comment]
Hah I found an obvious bug just by thinking about writing a test.

File: templates/default/nginx.conf.erb (hunk 3 above)
Comment (anfleene) [inline templates/default/nginx.conf.erb]: yes...fixed

File: test/integration/maintenance/serverspec/nginx_spec.rb (hunk 2 above)
Comment (bbergstrom) [inline test/integration/maintenance/serverspec/nginx_spec.rb]: I feel like I hit something similar the last time I worked with this cookbook.  I think I removed the failing test and moved on with my life.

**Comment by bbergstrom** [comment]
Looks like the config verification test needs to be updated for the log format name addition.

**Comment by anfleene** [comment]
Yeah I got side tracked by other stuff. This test run should pass.

**Comment by anfleene** [comment]
@bbergstrom Want to make one more final pass over this?

**Comment by bbergstrom** [comment]
:shipit:

**Pull request merged**